tests/test_exceptions.py

```diff
@@ -50,13 +50,23 @@ def test_python_call_in_catch():
     assert d["good"] is True
 
 
+if hasattr(pytest, "PytestUnraisableExceptionWarning"):  # Python >= 3.8 and pytest >= 6
```

Comment:
Alternatively, we move this into the function, and directly use the `warnings` module. But I'm not sure whether such a filter would get reset after the test or not, this way.